test/utils/floating_point_utils.hpp

```diff
@@ -0,0 +1,81 @@
+// Fizzy: A fast WebAssembly interpreter
+// Copyright 2020 The Fizzy Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+#pragma once
+
+#include <cmath>
+#include <cstdint>
+#include <limits>
+#include <type_traits>
+
+namespace fizzy::test
+{
+/// Simple implementation of C++20's std::bit_cast.
+template <typename DstT, typename SrcT>
+DstT bit_cast(SrcT x) noexcept
+{
+    DstT z;
+    static_assert(sizeof(x) == sizeof(z));
+    __builtin_memcpy(&z, &x, sizeof(x));
+    return z;
+}
+
+
+template <typename T>
+struct FP
+{
+    static_assert(std::is_floating_point_v<T>);
+
+    /// Shortcut to numeric_limits.
+    using limits = std::numeric_limits<T>;
```

Comment:
Why not `Limits`, `UintType`? I think it would be easier to see, that these are types.

Reply:
Can be. It just looks little weird when mixing with stuff from `std::numeric_limits`. Can constants stay lowercase?

Reply:
Yes, my preference would be CamelCase for types, lowercase for constants is fine

Reply:
Done.